python/taichi/examples/simulation/snow_phaseField.py

```diff
@@ -0,0 +1,222 @@
+                self.gamma * (self.temperature_equi - temperature[i, j]), 1.)
+            chemicalForce = phi[i, j] * (1. - phi[i, j]) * (phi[i, j] - 0.5 +
+                                                            m_chem)
+            gradForce_term1 = self.divergence_dEnergy_dGrad_term1(i, j)
+            grad_epsilon2 = ti.Vector([
+                (epsilons[ip, j]**2 - epsilons[im, j]**2) / (2. * dx),
+                (epsilons[i, jp]**2 - epsilons[i, jm]**2) / (2. * dx),
+            ])
+            grad_phi = ti.Vector([(phi[ip, j] - phi[im, j]) / (2. * dx),
+                                  (phi[i, jp] - phi[i, jm]) / (2. * dx)])
+            gradForce_term2 = grad_epsilon2[0] * grad_phi[0] + \
+                    grad_epsilon2[1] * grad_phi[1] + \
+                    epsilons[i, j]**2 * lapla_phi
+
+            self.phiRate[i, j][rk_loop] = mobility * (
+                chemicalForce + gradForce_term1 + gradForce_term2)
+            self.temperatureRate[
+                i,
+                j][rk_loop] = lapla_tp + self.latent_heat_coef * self.phiRate[
+                    i, j][rk_loop]
+
+    @ti.kernel
+    def rk4_total_update(self, ):
+        """the final step in RK4 (Runge-Kutta) process"""
+        dt, phi, phi_old, temperature, temperature_old, phiRate, temperatureRate = ti.static(
+            self.dt, self.phi, self.phi_old, self.temperature,
+            self.temperature_old, self.phiRate, self.temperatureRate)
+        for I in ti.grouped(phi):
+            for k in ti.static(range(4)):
+                phi_old[
+                    I] = phi_old[I] + self.weights_rk4[k] * dt * phiRate[I][k]
+                temperature_old[I] = temperature_old[
+                    I] + self.weights_rk4[k] * dt * temperatureRate[I][k]
+        for I in ti.grouped(phi):
+            phi[I] = phi_old[I]
+            temperature[I] = temperature_old[I]
+
+    @ti.func
+    def divergence_dEnergy_dGrad_term1(self, i, j):
+        im, jm, ip, jp = self.neighbor_index(i, j)
+        return (self.dEnergy_dGrad_term1[ip, j][0] - self.dEnergy_dGrad_term1[im, j][0]) / (2. * self.dx) + \
+            (self.dEnergy_dGrad_term1[i, jp][1] - self.dEnergy_dGrad_term1[i, jm][1]) / (2. * self.dx)
+
+    def advance(self, ):  # advance a time step
+        self.get_rate(rk_loop=0)
+        for rk_loop in range(1, 4):
+            self.rk4_intermediate_update(rk_loop=rk_loop)
+            self.get_rate(rk_loop=rk_loop)
+        self.rk4_total_update()
+
+    def getDendritic(self, steps=2048):
+        self.initialize()
+        gui_phi = ti.GUI("phase field", res=(self.n, self.n))
+
+        if self.writeImages:
+            path = "./results/"
+            if not os.path.exists(path): os.makedirs(path)
+
+        for i in range(steps):
+            if i % self.showFrameFrequency == 0:
+                gui_phi.set_image(self.phi)
+                gui_phi.show(path +
+                             "time_{:.4f}s.png".format(i * self.dt) if i %
```

Comment:
Could you please help make it a f-string like `f"time_{i * self.dt:.4f}s.png"` as the pre-commit requires?

Reply:
yeah, I will change it